gadget/update.go

```diff
@@ -336,13 +336,13 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 	if opts == nil {
 		opts = &VolumeCompatibilityOptions{}
 	}
-	logger.Debugf("checking volume compatibility between gadget volume %s and disk %s",
-		gadgetVolume.Name, diskVolume.Device)
+	logger.Debugf("checking volume compatibility between gadget volume %s (partial: %v) and disk %s",
+		gadgetVolume.Name, gadgetVolume.Partial, diskVolume.Device)
 
-	eq := func(ds *OnDiskStructure, vss []VolumeStructure, vssIdx int) (bool, string) {
-		gv := &vss[vssIdx]
+	eq := func(ds *OnDiskStructure, gv *Volume, vssIdx int) (bool, string) {
```

Comment:
do we need the the change of signature? and if we do, do we need gv to be passed in? as per my other comment